src/gmt_vector.c

```diff
@@ -446,7 +446,7 @@ GMT_LOCAL uint64_t gmtvector_fix_up_path_cartesian (struct GMT_CTRL *GMT, double
 	if (step <= 0.0) step = 1.0;	/* Sanity valve; if step not given we set it to 1 */
 
 	for (i = 1; i < n; i++) {
-		if (mode == GMT_STAIRS_Y) {	/* First follow x, then y */
+		if (mode == GMT_STAIRS_X) {	/* First follow x, then y */
```

Comment:
This `if`-statement should check `mode == GMT_STAIR_X`, because at lines 455, the y values are constant and the x values are changing.

Reply:
Yes, and with GMT_STAIRS_Y = 1, GMT_STAIRS_X = 2, that matches now all the comments.
As a matter of fact, I met this issue once, but thought I had made a user error.